### PR TITLE
Add attributes 'title_when' and 'title_depth'

### PR DESCRIPTION
**Pull request by t-wissmann**
This makes it configurable when to show the title bar. So far, there are
four options that essentially depend on the number of tabs.
So far, I've used padding_top to add space for characters in the title
that reach below the baseline. Since the space below the baseline should
be present only when the title is shown, I've added a new setting
title_depth.

This new feature was requested multiple times on the IRC and in the
discussion of #1097 and #1388.

### CODE REVIEW

**Comment by t-wissmann** [comment]
I can fix the failing test case by extending `showTitle()` with:
```C++
bool DecorationScheme::showTitle(size_t tabCount) const
{
    if (title_height() == 0) {
        return false;
    }
    switch ...
}
```
but first I'd like to understand where X11 errors come from that make hlwm quit.

File: tests/test_theme.py [hunk 1]
```diff
@@ -134,3 +136,79 @@ def test_decoration_geometry_vs_x11_data(hlwm, x11, floating):
...
+                assert title_size == expected_title_size
+            else:
+                assert title_size == 0
+
+
+@pytest.mark.parametrize("floating", [True, False])
+def test_title_when_for_absence_of_tabs(hlwm, floating):
+    """
+    for floating clients or frames with an algorithm other than 'max',
+    there should no be tabs at all and thus the titlebar is
```
Comment (blarz) [inline tests/test_theme.py:189]: be no

**Review by blarz — CHANGES_REQUESTED** [review]
Just a small grammar error, otherwise looks very good!

File: tests/test_theme.py (hunk 1 above)
Comment (blarz) [reply tests/test_theme.py:189]: or not be ;)

Comment (t-wissmann) [reply tests/test_theme.py]: oups. yes, "there should be no tabs at all" :-)

**Comment by t-wissmann** [comment]
Thanks :)

**Review by blarz — APPROVED** [review]
Very cool, thank you! :)

**Pull request merged**